node/pkg/raft/raft.go

```diff
@@ -60,28 +62,34 @@ func (r *Raft) subscribe(ctx context.Context) {
 		log.Error().Err(err).Msg("failed to subscribe to topic")
 	}
 	for {
+		if ctx.Err() == context.Canceled {
```

Comment:
It is maybe too late to say this, but isn't there a solution without an infinite loop? 👀

Reply:
🤔 can you give me more detail? it should continously get message from other raft peers so I put it inside infinite poop

Reply:
I thought it was blocking call sonce the official example code also implemented it through infinite for loop

'''
func printMessagesFrom(ctx context.Context, sub *pubsub.Subscription) {
	for {
		m, err := sub.Next(ctx)
		if err != nil {
			panic(err)
		}
		fmt.Println(m.ReceivedFrom, ": ", string(m.Message.Data))
	}
}
'''
https://github.com/libp2p/go-libp2p/blob/master/examples/pubsub/basic-chat-with-rendezvous/main.go

Reply:
Got it! So the `subscribe` function is called through go routine?

Reply:
yes, you can refer to line 40

Reply:
Thank you! I guess it is good to keep, but I am surprised they haven't implement it with channels instead.

Reply:
🤔 i'll double check official docs tmr